performant-software/Annotation-Studio
# Add document folder feature

# What this PR does

This PR adds a basic "document folder" feature:

1. Anthology owners can apply tags to documents.
2. Anthology members can filter by those tags.
3. Tag filtering is done with a query string, so anthology owners can link students directly to a folder.

## Screenshots

![Screen Shot 2022-08-30 at 12 40 28 PM](https://user-images.githubusercontent.com/64725469/187492637-fb444f40-2ed5-410b-b986-d8141203e665.png)

![Screen Shot 2022-08-30 at 12 40 48 PM](https://user-images.githubusercontent.com/64725469/187492691-acc3cccb-2200-4aa9-be5f-b81efcfe9720.png)

# How to test

1. Visit https://staging.covecollective.org
2. Open an anthology with documents in it (if there isn't one, create one and add documents).
3. Click on the "Folders" tab (see my first screenshot)
4. You should **not** see the Folders column that is visible in my screenshot.
5. Select a few documents and add them to a new folder with the "Add to new folder" text box.
6. The new folder's name should now appear as an option in the "Add to folder" dropdown (see my second screenshot).
7. Add 1 or 2 more folder names, to a mix of documents, including some of the ones you tagged in step 5, so that there's an assortment of folder combinations.
8. In the "Folder" column, each folder name should have an X on its righthand side. Click one to remove the corresponding document from that folder.
9. Click the text of a folder name and a) the list of documents should only show documents in that folder, and b) the table header should display the name of the folder, such as:
![Screen Shot 2022-08-30 at 12 48 36 PM](https://user-images.githubusercontent.com/64725469/187494255-3122586f-9d33-441f-8b0e-907a9eca0652.png)
10. Create a new anthology and add some of the same documents that were in your first one.
11. There shouldn't be any folders in this anthology.
